pkg/istio/onboarding/controller.go

```diff
@@ -180,14 +180,30 @@ func (c *Controller) getOnboardedServiceList() []string {
 	return serviceList
 }
 
-// errHandler re-adds the key for a failed processor.sync operation
-func (c *Controller) errHandler(err error, item *processor.Item) error {
-	if err != nil {
-		if item != nil {
-			log.Errorf("Error performing %s on %s: %s", item.Operation, item.Resource.Key(), err.Error())
-		}
-		c.queue.AddRateLimited(queueKey)
+// callbackHandler re-adds the key for a failed processor.sync operation
+func (c *Controller) callbackHandler(err error, item *processor.Item) error {
+	if err == nil {
+		return nil
+	}
+	if item != nil {
+		log.Errorf("Error performing %s on %s: %s", item.Operation, item.Resource.Key(), err)
+	}
+	if apiErrors.IsNotFound(err) || apiErrors.IsAlreadyExists(err) {
+		log.Infof("Error is non-retryable %s", err)
+		return nil
```

Comment:
Shouldn't this trigger a new resync of the controller?

Reply:
Similar logic goes here.. since the change already went through by another operation, and that would anyway trigger another sync, we do not have to retry/re-add this error

Reply:
That makes sense, thanks.